src/Microsoft.DotNet.ProjectModel/ProjectContextBuilder.cs

```diff
@@ -277,7 +277,7 @@ public ProjectContext Build()
 
                     if (platformDependency != null)
                     {
-                        platformLibrary = libraries[new LibraryKey(platformDependency.Value.Name)];
+                        libraries.TryGetValue(new LibraryKey(platformDependency.Value.Name), out platformLibrary);
```

Comment:
I'm thinking in terms of usage of the projectmodel api using exceptions would make error cases more visible and bug fixes faster. 

If we need invalid ProjectContext support you are completely right though and I don't know the answer to that question.